### ISSUE TITLE
SQS list queue tags outputs unexpected format with -text flag

### ISSUE THREAD

**Issue by ziyanli-amazon** [aws/aws-cli#7839]
### Describe the bug

This is SQS team, we recently got a customer reporting unexpected output format when using `--output text` flag.
When querying `list-queue-tags` command without any flag, the output is as expected
```
aws sqs list-queue-tags --queue-url https://sqs.us-east-1.amazonaws.com/xxxxxxxxxxx/DLQ
{
    "Tags": {
        "hi": "test2",
        "hello": "test"
    }
}
```

When using the `--output -text` flag, the output becomes
```
aws sqs list-queue-tags --queue-url https://sqs.us-east-1.amazonaws.com/xxxxxxxxxxxxx/DLQ --output text
TAGS    test    test2
```
It only displays the values of each tag.

This behavior is reproducible and we suspect this bug comes from [this line](https://github.com/aws/aws-cli/blob/develop/awscli/text.py#L20). 

### Expected Behavior

```
aws sqs list-queue-tags --queue-url https://sqs.us-east-1.amazonaws.com/xxxxxxxxxxx/DLQ --output text
{
    "Tags": {
        "hi": "test2",
        "hello": "test"
    }
}
```

'Tags': [{'Key': 'testKey2', 'Value': 'testValue2'}, {'Key': 'testKey', 'Value': 'testValue'}]

### Current Behavior

```
aws sqs list-queue-tags --queue-url https://sqs.us-east-1.amazonaws.com/xxxxxxxxxxxxx/DLQ --output text
TAGS    test    test2
```

### Reproduction Steps

```
aws sqs list-queue-tags --queue-url https://sqs.us-east-1.amazonaws.com/xxxxxxxxxxxxx/DLQ --output text
```

### Possible Solution

https://github.com/aws/aws-cli/blob/develop/awscli/text.py#L20
Check if the response contains a Tags attribute, and that Tags attribute is not formatted as a list of dictionaries, then change it

### CLI version used

2.11.11

### Environment details (OS name and version, etc.)

MacOS

**Comment by tim-finnigan** [comment]
Hi @ziyanli-amazon here is the documentation for `--output text`: https://docs.aws.amazon.com/cli/latest/userguide/cli-usage-output-format.html#text-output

As shown in the example for `aws iam list-users --output text`, it will display a list of `USERS` and the values (not the keys) for each item in that list. Based on that documentation, what you're describing appears to be the expected behavior.

I think most service APIs return Tags in a list rather than dict. For example with EC2 you could run a command like `aws ec2 describe-volumes --query "Volumes[*].Tags" --output text` to output the tags in a Key / Value format because the `Tags` are structured like this:

```
"Tags": [
    {
        "Key": "tag1",
        "Value": "value1"
    },
    {
        "Key": "tag2",
        "Value": "value2"
    }
],
``` 

As a workaround for your example you could do something like `aws sqs list-queue-tags --queue-url <url> --query "Tags.{hi: [hi], hello: [hello]}" --output text`, which isn't ideal but maybe there are better approaches. Also the SQS team could consider adding the ability to update the output structure of Tags for compatibility with `--output text`.

To summarize, what you're describing appears to be expected behavior based on the documentation and the Tags shape that SQS returns. But we can consider this to be a feature request for handling these types of use cases. If others are similarly affected, they can 👍 the issue and share their use case.

**Comment by ziyanli-amazon** [comment]
@tim-finnigan Thanks for the detailed explanation and diving deep. I've inspected our Api Model file and we are indeed defining the tag attribute type as `map/dictionary`. And changing the type to `list` will introduce backwards incompatibility for other APIs like `TaggingOnCreateQueue`, `TagQueue`, and `UntagQueue`. 

I saw you've changed the category of the issue i raised to `feature-request`. I really appreciate you and the aws cli team role modelling Customer Obsession. I've provided a possible solution to it in this Issue, but I'm sure there are better solutions. Looking forward to seeing such feature gets released soon.
